src/cli/traceViewer/web/ui/networkResourceDetails.tsx

```diff
@@ -0,0 +1,112 @@
+        setRequestBody(requestResource);
+      }
+
+      if (resource.responseSha1 !== 'none') {
+        const responseResource = await window.readResource(resource.responseSha1);
+        setResponseBody(responseResource);
+      }
+    };
+
+    readResources();
+  }, [expanded, resource.responseSha1, resource.requestSha1]);
+
+  function formatBody(body: string | null, contentType: string): string {
+    if (body === null)
+      return 'Loading...';
+
+    const bodyStr = atob(body);
+
+    if (contentType.includes('application/json')) {
+      try {
+        return JSON.stringify(JSON.parse(bodyStr), null, 2);
+      } catch (err) {
+        return bodyStr;
+      }
+    }
+
+    if (contentType.includes('application/x-www-form-urlencoded'))
+      return decodeURIComponent(bodyStr);
+
+    return bodyStr;
+  }
+
+  function formatStatus(status: number): string {
+    if (status >= 200 && status < 400)
+      return 'status-success';
+
+    if (status >= 400)
+      return 'status-failure';
+
+    return 'status-neutral';
+  }
+
+  const requestContentTypeHeader = resource.requestHeaders.find(q => q.name === 'Content-Type');
+  const requestContentType = requestContentTypeHeader ? requestContentTypeHeader.value : '';
+
+  return <div
+    className={'network-request ' + (selected ? 'selected' : '')} onClick={() => setSelected(index)}>
+    <Expandable expanded={expanded} setExpanded={setExpanded} style={{ width: '100%' }} title={
+      <div className='network-request-title'>
+        <div className={'network-request-title-status ' + formatStatus(resource.status)}>{resource.status}</div>
+        <div className='network-request-title-method'>{resource.method}: &nbsp;</div>
+        <div className='network-request-title-url'>{resource.url}</div>
+        <div className='network-request-title-content-type'>{resource.contentType}</div>
+      </div>
+    } body={
+      <div className='network-request-details'>
+        <h4>URL</h4>
+        <div className='network-request-details-url'>{resource.url}</div>
+        <h4>Request Headers</h4>
+        <div className='network-request-headers'>{resource.requestHeaders.map(pair => `${pair.name}: ${pair.value}`).join('\n')}</div>
+        <h4>Response Headers</h4>
+        <div className='network-request-headers'>{resource.responseHeaders.map(pair => `${pair.name}: ${pair.value}`).join('\n')}</div>
+        {resource.requestSha1 !== 'none' ? <h3>Request Body</h3> : ''}
+        {resource.requestSha1 !== 'none' ? <div className='network-request-body'>{formatBody(requestBody, requestContentType)}</div> : ''}
+        <h4>Response Body</h4>
+        {responseBody !== null && resource.contentType.includes('image') ? <img src={`data:${resource.contentType};base64,${responseBody}`} /> : ''}
+        {responseBody !== null && !resource.contentType.includes('image') ? <div className='network-request-response-body'>{formatBody(responseBody, resource.contentType)}</div> : ''}
```

Comment:
I do see a lot of response bodies being empty (for xhrs). Let's do `<empty>` when it is empty, and `<unavailable>` when it is null?

Reply:
I went for something more descriptive for the `<unavailable>` case, what do you think?